src/triage/component/catwalk/predictors.py

```diff
@@ -159,6 +171,89 @@ def _write_to_db(
 
+    def _write_metadata_to_db(self, model_id, matrix_store, postgres_random_seed, save_predictions):
+        orm_obj = matrix_store.matrix_type.prediction_metadata_obj(
+            model_id=model_id,
+            matrix_uuid=matrix_store.uuid,
+            tiebreaker_ordering=self.rank_order,
+            random_seed=postgres_random_seed,
+            predictions_saved=save_predictions,
+        )
+        session = self.sessionmaker()
+        session.merge(orm_obj)
+        session.commit()
+        session.close()
+        
+    def update_db_with_ranks(self, model_id, matrix_uuid, prediction_obj, postgres_random_seed):
+        """Update predictions table with rankings, both absolute and percentile.
+        All entities should have different ranks, so to break ties:
+        - abs_rank uses the 'row_number' function, so ties are broken by the database ordering
+            session.close()
+        - pct_rank uses the output of the abs_rank to compute percentiles
+          (as opposed to raw scores), so it inherits the tie-breaking from abs_rank
+        Args:
+            model_id (int) the id of the model associated with the given predictions
+            matrix_uuid (string) the uuid of the prediction matrix
+        """
+        logging.info(
+            'Beginning ranking of new Predictions for model %s, matrix %s',
+            model_id,
+            matrix_uuid
+        )
+        with self.db_engine.begin() as conn:
+            conn.execute(f'''
+                select setseed(%(postgres_random_seed)s);
+                with abs_ranks_no_ties as (
+                    select
+                        entity_id,
+                        as_of_date,
+                        row_number() over (
+                            partition by as_of_date order by score desc, {self.rank_order}
+                        ) as abs_rank
+                        from {table_name(prediction_obj)}
+                        where model_id = %(model_id)s and matrix_uuid = %(uuid)s
+                ), pct_ranks as (
+                    select
+                        entity_id,
+                        as_of_date,
+                        percent_rank() over (
+                            partition by as_of_date order by abs_rank desc
+                        ) as pct_rank
+                        from abs_ranks_no_ties
+                ), abs_ranks_with_ties as (
+                    select
+                        entity_id,
+                        as_of_date,
+                        rank() over (
+                            partition by as_of_date order by score desc
+                        ) as abs_rank
+                        from {table_name(prediction_obj)}
+                        where model_id = %(model_id)s and matrix_uuid = %(uuid)s
+                )
+                update {table_name(prediction_obj)} as p
+                set
+                    rank_abs = abs.abs_rank,
+                    rank_pct = pct.pct_rank,
```

Comment:
I realize it would mean changing column names that already exist, but I would probably lean towards naming these `rank_abs_no_ties` and `rank_pct_no_ties` (or `without_ties`) just to be very explicit as well as to avoid suggesting that one is the preferred/default to use in general.

Also, we might have to calculate it directly, but it seems like we may also want `rank_pct_with_ties` too?